CRM/Core/Permission.php

```diff
@@ -662,10 +662,17 @@ public static function getCorePermissions() {
       'view all contacts' => [
         'label' => $prefix . ts('view all contacts'),
         'description' => ts('View ANY CONTACT in the CiviCRM database, export contact info and perform activities such as Send Email, Phone Call, etc.'),
+        'implies' => [
+          'view my contact',
```

Comment:
I'm a bit worried about this one - historically the query for 'view my contact' contained an OR clause - which meant it performed terribly - it might be that having 'view all contacts' still avoid the bad query & this is just aesthetic

Reply:
@eileenmcnaughton I'm pretty sure this PR doesn't affect how queries are constructed. What this would do is:
1. Provide some metadata about what permissions fall under others (potentially for use in the Standalone Permissions UI).
2. Makes setting & checking permissions simpler. It's no longer necessary to check if a contact has either one or the other, now just check for the narrower permission ('view my') and it automatically includes anyone with the wider permission ('view all').

The only difference this makes at runtime is that previously it was possible to grant a user 'view all' and *not* grant them 'view my', but that's a nonsensical thing to do, or rather, it's nonsensical to expect that to have a different effect than granting both.

Looking at the code that does the server-hating query you mentioned, I don't think this patch affects it as the permission checks will end up returning the same result before & after:

https://github.com/civicrm/civicrm-core/blob/f704d628e1484cc9ca425970d951b7f482a31ce9/CRM/ACL/API.php#L110-L130

Reply:
@colemanw ok